### DISCUSSION TITLE
Extract origins of ingredients

### DISCUSSION THREAD

**Discussion by teolemon** [openfoodfacts/robotoff#1209] [Ideas]
### What

- Some geographical information is not in the ingredient list, but spread on the packaging
- Extract this information to detections, that can then be combined to create insights. 
- This will help get origins for the eco-score computation.
- They can be asked as the following Yes/No question: "Is the following statement valid ?"
### Triggers
  - “Importé du {Pérou} >> en:Ingredients from Peru
  - “{Quinoa} origine non-UE” (séparé de la liste des ingrédients) >> en:Quinoa from outside Europe
  - “Certains ingrédients ne proviennent pas de {France}” >>en:Non {{French}} ingredients
  - “à partir de blé dur de diverses origines” >> en:Wheat from various origins

### Part of 
- #600 
- #376

**Comment by senthilnathan-ra** [comment]
@teolemon I'm new to the community and want to contribute. Is this issue still open?

**Comment by Jagrutiti** [comment]
Hey @teolemon

How do I get the geographical information? Is there an API for it? Could you please shed more light on this? 

Thanks :)

**Comment by alexgarel** [comment]
@Jagrutiti informations are to be searched in OCR. 
There already are such kind of regexp based search in OCR. For example for nutrients : https://github.com/openfoodfacts/robotoff/blob/master/robotoff/prediction/ocr/nutrient.py (and the folder contains more).

It's something like building regular expressions (in diverse language) expecting a country in it (we then have a taxonomy for countries, so you can get all known countries from there : https://static.openfoodfacts.net/data/taxonomies/countries.full.json, such file can be uploaded by robotoff through a scheduler method)
